src/odb/BUILD

```diff
@@ -0,0 +1,113 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2025-2025, Precision Innovations Inc.
+
+load("//:bazel/tcl_encode_or.bzl", "tcl_encode")
+load("//:bazel/tcl_wrap_cc.bzl", "tcl_wrap_cc")
+
+package(
+    default_visibility = ["//:__subpackages__"],
+    features = ["layering_check"],
+)
+
+filegroup(
+    name = "swig_imports",
+    srcs = glob([
+        "src/swig/tcl/*.i",
+        "src/swig/common/*.i",
+    ]),
+)
+
+cc_library(
+    name = "odb",
+    srcs = glob([
+        "src/cdl/*.cpp",
+        "src/db/*.cpp",
+        "src/db/*.h",
+        "src/db/*.hpp",
+        "src/zutil/*.cpp",
+        "src/defout/*.cpp",
+        "src/defout/*.h",
+        "src/defin/*.cpp",
+        "src/defin/*.h",
+        "src/lefin/*.cpp",
+        "src/lefin/*.h",
+        "src/lefout/*.cpp",
+        "src/swig/tcl/MakeOdb.cpp",
+        "src/swig/common/swig_common.cpp",
+        "src/swig/common/swig_common.h",
+    ]),
+    hdrs = glob([
+        "include/odb/*.h",
+        "include/odb/*.hpp",
+    ]),
+    copts = [
+        "-fexceptions",
+        "-Wno-error",
+    ],
+    features = [
+        "-use_header_modules",
+    ],
+    includes = [
+        "include",
+    ],
+    deps = [
+        "//src/odb/src/def",
+        "//src/odb/src/def:defzlib",
+        "//src/odb/src/lef",
+        "//src/odb/src/lef:lefzlib",
+        "//src/utl",
+        "@boost.algorithm",
+        "@boost.bind",
+        "@boost.config",
+        "@boost.fusion",
+        "@boost.geometry",
+        "@boost.lambda",
+        "@boost.optional",
+        "@boost.phoenix",
+        "@boost.polygon",
+        "@boost.property_tree",
+        "@boost.regex",
+        "@boost.spirit",
+        "@spdlog",
+        "@tk_tcl//:tcl",
+        "@zlib",
+    ],
+)
+
+tcl_wrap_cc(
+    name = "swig",
+    srcs = glob([
+        "src/swig/tcl/*.i",
```

Comment:
the first two glob statements are already included in the `:swig_imports`, so you can probably minimize that.

Reply:
Good point.  I'll fix that in a follow on.